include/modem/at_parser.h

```diff
@@ -0,0 +1,307 @@
+ * @retval -EBADMSG    The AT command string is malformed.
+ */
+int at_parser_int32_get(struct at_parser *parser, size_t index, int32_t *value);
+
+/**
+ * @brief Get a subparameter value as an unsigned 32-bit integer number.
+ *
+ * @param[in]  parser AT parser.
+ * @param[in]  index  Subparameter index in the current AT command line configured in @p parser.
+ * @param[out] value  Subparameter value.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ * @retval -EINVAL     One or more of the supplied parameters are invalid.
+ * @retval -EPERM      @p parser has not been initialized.
+ * @retval -EOPNOTSUPP Operation not supported for the subparameter type at the given index.
+ * @retval -ERANGE     Parsed integer value is out of range for the expected type.
+ * @retval -EBADMSG    The AT command string is malformed.
+ */
+int at_parser_uint32_get(struct at_parser *parser, size_t index, uint32_t *value);
+
+/**
+ * @brief Get a subparameter value as a signed 64-bit integer number.
+ *
+ * @param[in]  parser AT parser.
+ * @param[in]  index  Subparameter index in the current AT command line configured in @p parser.
+ * @param[out] value  Subparameter value.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ * @retval -EINVAL     One or more of the supplied parameters are invalid.
+ * @retval -EPERM      @p parser has not been initialized.
+ * @retval -EOPNOTSUPP Operation not supported for the subparameter type at the given index.
+ * @retval -ERANGE     Parsed integer value is out of range for the expected type.
+ * @retval -EBADMSG    The AT command string is malformed.
+ */
+int at_parser_int64_get(struct at_parser *parser, size_t index, int64_t *value);
+
+/**
+ * @brief Get a subparameter value as an unsigned 64-bit integer number.
+ *
+ * @param[in]  parser AT parser.
+ * @param[in]  index  Subparameter index in the current AT command line configured in @p parser.
+ * @param[out] value  Subparameter value.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ * @retval -EINVAL     One or more of the supplied parameters are invalid.
+ * @retval -EPERM      @p parser has not been initialized.
+ * @retval -EOPNOTSUPP Operation not supported for the subparameter type at the given index.
+ * @retval -ERANGE     Parsed integer value is out of range for the expected type.
+ * @retval -EBADMSG    The AT command string is malformed.
+ */
+int at_parser_uint64_get(struct at_parser *parser, size_t index, uint64_t *value);
+
+/**
+ * @brief Get a value as a string.
+ *
+ * The data type must be a string (quoted or non-quoted), an AT command prefix, or an array,
+ * otherwise an error is returned.
+ * The string value is copied to the buffer.
+ * @p len must be bigger than the string length, or an error is returned.
+ * The copied string is not null-terminated.
```

Comment:
Why is the string not null-terminated? I think it would be quite convenient to have the string be null-terminated, so that it's ready to be printed or otherwise manipulated with `string.h` functions.

Reply:
Re-added null termination.